client/state/data-layer/utils.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * External dependencies
+ */
+import {
+	isArray,
+	mergeWith,
+} from 'lodash';
+
+const concatHandlers = ( left, right ) =>
+	isArray( left )
+		? left.concat( right )
+		: undefined;
+
+export const mergeHandlers = ( ...handlers ) =>
+	handlers.length > 1
+		? mergeWith( {}, ...handlers, concatHandlers )
+		: handlers.shift();
```

Comment:
>I just prefer `.shift()` in many cases is all.

Perhaps I'm sensitive to it because it sticks out as a mutation, when in fact you don't care about the mutation, you're simply referencing the first entry.

Also worth pointing out that `Array#shift` can be very slow ([stackoverflow](http://stackoverflow.com/questions/6501160/why-is-pop-faster-than-shift), p4TIVU-4Ft-p2)